test/e2e/autoscale.go

```diff
@@ -336,16 +340,27 @@ func checkPodScale(ctx *testContext, targetPods, minPods, maxPods float64, durat
 	}
 }
 
-func assertAutoscaleUpToNumPods(ctx *testContext, curPods, targetPods float64, duration time.Duration, quick bool) {
+// assertAutoscaleUpToNumPods supports two test modes: quick, and not quick.
+// 1) Quick mode: succeeds when the number of pods meets targetPods.
+// 2) Not Quick (sustaining) mode: succeeds when the number of pods gets scaled to targetPods and
+//    sustains there for the `duration`.
+func assertAutoscaleUpToNumPods(ctx *TestContext, curPods, targetPods float64, duration time.Duration, quick bool) {
+	assertAutoscaleUpToNumPodsWithDurationAndDone(ctx, curPods, targetPods, duration, time.After(duration), quick)
+}
+
+// AssertAutoscaleUpToNumPodsWithDone asserts the number of pods gets scaled to targetPods and
+// sustains there until the `done` channel sends a signal.
+func AssertAutoscaleUpToNumPodsWithDone(ctx *TestContext, curPods, targetPods float64, done <-chan time.Time) {
+	// We use 10 hours here which should be suffient for the e2e tests.
+	assertAutoscaleUpToNumPodsWithDurationAndDone(ctx, curPods, targetPods, 10*time.Hour, done, false /* quick */)
+}
+
+// DO NOT USE this one directly. Use assertAutoscaleUpToNumPods or AssertAutoscaleUpToNumPodsWithDone instead.
```

Comment:
Why not?

Reply:
well, it's kind of internal implementation. But sure if someone wants to use this directly for some case. Updated the documents to include conditions.